src/libutil/url.cc

```diff
@@ -183,4 +184,20 @@ std::string fixGitURL(const std::string & url)
     }
 }
 
+// https://www.rfc-editor.org/rfc/rfc3986#section-3.1
+bool isValidSchemeName(std::string_view s)
+{
```

Comment:
This should just check against the `schemeRegex` already defined in `url-parts.hh` (which is based on RFC 3986). That also removes the need for the `isASCII*` functions.

Reply:
Didn't see that. Only moved the code to this file when tidying up my initial implementation. Will use the regex.
I'll keep the ASCII functions around on a branch for use in other places where contributors otherwise come up with cleanup suggestions that aren't quite as good.

Reply:
It's not actually compliant. We ought to accept upper case (like I did) and then convert to lower case (which neither of us did). It's safer now, but not compliant.

Reply:
Made it compliant.

Reply:
Not quite yet. Scoped that out because URLs haven't (always?) been parsed and canonicalized at this stage. Better to reject upper case for the time being.